app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,89 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id]
+
+    @task = Task.find_by(id: task_id)
+
+    unless @task
+      redirect_to tasks_path, :flash => {:error => "Could not find task with id: #{task_id}"}
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def edit
+    task_id = (params[:id])
+
+    @task = Task.find_by(id: task_id)
+
+    unless @task
+      redirect_to tasks_path, :flash => {:error => "Could not find task with id: #{task_id}"}
+    end
+  end
+
+  def update
+    task_id = (params[:id])
+
+    @task = Task.find_by(id: task_id)
+
+    if @task
+      @task.name = params[:name]
+      @task.description = params[:description]
```

Comment:
This doesn't match how the name and description are stored in `params`. The result is that both `name` and `description` get set to `nil` here.

This is why manual testing is important in addition to automated testing - if the tests are wrong, your human eyeballs are the only way to catch the problem.